core/src/main/java/org/apache/calcite/sql/dialect/PhoenixSqlDialect.java

```diff
@@ -40,4 +47,19 @@ public PhoenixSqlDialect(Context context) {
   @Override public boolean supportsCharSet() {
     return false;
   }
+
+  @Override public @Nullable SqlNode getCastSpec(RelDataType type) {
+    switch (type.getSqlTypeName()) {
+    case REAL:
+      return new SqlDataTypeSpec(
+          new SqlAlienSystemTypeNameSpec(
+              "FLOAT",
+              type.getSqlTypeName(),
+              SqlParserPos.ZERO),
+          SqlParserPos.ZERO);
+    default:
+      break;
+    }
+    return super.getCastSpec(type);
```

Comment:
Can't you handle arrays, multisets, and maps here?

Reply:
Had handled with array and map types in cast logic，because Hive/Phonenix had not type of multiset,  PTAL @mihaibudiu